src/hooks/sqlite.ts

```diff
@@ -0,0 +1,125 @@
+import assert from "assert";
+
+import sqlite from "sqlite3";
+
+import { recording } from "../recorder";
+import { getTime } from "../util/getTime";
+
+type RecordingProxyTarget =
+  | typeof sqlite.Database.prototype.run
+  | typeof sqlite.Database.prototype.all
+  | typeof sqlite.Database.prototype.get
+  | typeof sqlite.Database.prototype.each
+  | typeof sqlite.Database.prototype.exec
+  | typeof sqlite.Statement.prototype.run
+  | typeof sqlite.Statement.prototype.all
+  | typeof sqlite.Statement.prototype.get
+  | typeof sqlite.Statement.prototype.each;
```

Comment:
What about `map`?

Reply:
I saw it in the documentation [here]( https://github.com/TryGhost/node-sqlite3/wiki/API#mapsql--callback-1) but I couldn't see it in the type declaration in @types/sqlite3/index.d.ts. I'll try to be sure that it exists and do something like:

```
// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-explicit-any  
  (mod.Statement.prototype as any).map = createRecordingProxy(
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-explicit-any  
    (mod.Statement.prototype as any).map as RecordingProxyTarget,
  );
```

Reply:
Nevermind, it actually calls `all` under the hood: https://github.com/TryGhost/node-sqlite3/blob/44e570aff823d6626d30558c8057a12a27476cf6/lib/sqlite3.js#L144